### PR TITLE
[AIRFLOW-XXX] Fix display of SageMaker operators/hook docs

### PR DESCRIPTION
**Pull request by ashb**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

There were a number of minor formatting glitches that meant these didn't
display 100% right once written to HTML. I use "intersphinx" ext to link directly to the boto3 class-names by name, rather than URL.

**Before**
screen shot 2018-12-01 at 23 30 57

**After**
screen shot 2018-12-01 at 23 30 31

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/hooks/sagemaker_hook.py [hunk 1]
```diff
@@ -253,10 +253,11 @@ def log_stream(self, log_group, stream_name, start_time=0, skip=0):
         :param skip: The number of log entries to skip at the start (default: 0).
             This is for when there are multiple entries at the same timestamp.
         :type skip: int
-        :return:A CloudWatch log event with the following key-value pairs:
-            'timestamp' (int): The time in milliseconds of the event.
-            'message' (str): The log event data.
-            'ingestionTime' (int): The time in milliseconds the event was ingested.
+        :rtype: dict
+        :return: | A CloudWatch log event with the following key-value pairs:
+                 |   'timestamp' (int): The time in milliseconds of the event.
```
Comment (kaxil) [inline airflow/contrib/hooks/sagemaker_hook.py:258]: what does the pipe (`|`) do here?

Comment (ashb) [reply airflow/contrib/hooks/sagemaker_hook.py:258]: It's called a "line block" in the rst docs i think. Preserves new lines and leading spaces in this case. Without this it combines all in to one line

Comment (kaxil) [reply airflow/contrib/hooks/sagemaker_hook.py:258]: Oh, cool. Good to know.

**Comment by kaxil** [comment]
Cherry-picked it into 1.10.2 as well.

**Pull request merged**